pgcli/pgexecute.py

```diff
@@ -215,6 +221,9 @@ def connect(self, database=None, user=None, password=None, host=None,
         self.host = host
         self.port = port
 
+        if not self.host:
+            self.host = self.socket_directory()
```

Comment:
When calling `pgcli with -h /tmp` it does exactly that.
As for reconnect, I came across a problem (even on master) as when I restart the database `psycopg2.OperationalError.pgcode` is `None`. This wasn't handled resulting in a `KeyError` exception within psycopg2.

Now everything is working fine.